app/models/subnet.rb

```diff
@@ -396,6 +409,13 @@ def ensure_ip_addrs_valid
     end
   end
 
+  def ensure_gateway_in_subnet
+    gw = IPAddr.new(gateway)
+    errors.add(:gateway, _("is not from the subnet")) unless masked_subnet_address.include?(gw)
```

Comment:
this makes "out of subnet" gateways as @ekohl showed in his example impossible.

I'm personally fine with being that strict, but Ewoud might think differently :)

Reply:
As noted elsewhere, I certainly don't think it belongs in this PR because users may be surprised. As its own PR, we can discuss the impact.

Reply:
Yup, I did not think it through. We do not totally enforce that, makes no sense.